wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditReportPrinter.java

```diff
@@ -0,0 +1,151 @@
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * .
+     * .
+     * .
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * <p>
+     * If the list of user workspaces is not defined it prints error message.
+     *
+     * @param auditReport
+     *         file of audit report
+     * @param user
+     *         information about user collected in {@link UserImpl} object
+     * @param workspaces
+     *         list of workspaces that are related to given user
+     * @param wsPermissions
+     *         map of permissions to workspaces
+     * @throws ServerException
+     *         if an error occurs
+     */
+    void printUserInfoWithHisWorkspacesInfo(Path auditReport,
+                                            UserImpl user,
+                                            List<WorkspaceImpl> workspaces,
+                                            Map<String, AbstractPermissions> wsPermissions) throws ServerException {
+        long permissionsNumber = wsPermissions.values().stream().filter(permissions -> permissions != null).count();
+        long ownWorkspacesNumber = workspaces.stream().filter(workspace -> workspace.getNamespace().equals(user.getName())).count();
+        printRow(user.getEmail() + " is owner of " +
+                 ownWorkspacesNumber + " workspace" + (ownWorkspacesNumber > 1 | ownWorkspacesNumber == 0 ? "s" : "") +
+                 " and has permissions in " + permissionsNumber + " workspace" +
+                 (permissionsNumber > 1 | permissionsNumber == 0 ? "s" : "") + "\n", auditReport);
+        for (WorkspaceImpl workspace : workspaces) {
+            printUserWorkspaceInfo(workspace, user, wsPermissions, auditReport);
+        }
+    }
+
+    /**
+     * Prints error in format:
+     * [ERROR] <error text>!
+     *
+     * @param error
+     *         text of error
+     * @param auditReport
+     *         file of audit report
+     * @throws ServerException
+     *         if an error occurs
+     */
+    void printError(String error, Path auditReport) throws ServerException {
+        printRow("[ERROR] " + error + "!\n", auditReport);
+    }
+
+    private void printUserWorkspaceInfo(WorkspaceImpl workspace, UserImpl user, Map<String, AbstractPermissions> wsPermissions,
+                                        Path auditReport) throws ServerException {
+        printRow("   └ " + workspace.getConfig().getName() +
+                 ", is owner: " + workspace.getNamespace().equals(user.getName()) + ", permissions: ", auditReport);
+        AbstractPermissions permissions = wsPermissions.get(workspace.getId());
+        if (permissions != null) {
+            printRow(permissions.getActions().toString() + "\n", auditReport);
+        } else {
+            printError("Failed to retrieve workspace permissions", auditReport);
```

Comment:
Are you sure that you should print this message when permissions were not found?